### PR TITLE
[ticket/17411] Add core event to search.php

### PR DESCRIPTION
**Pull request by rxu**
Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

<a href="https://tracker.phpbb.com/browse/PHPBB-17411">PHPBB-17411</a>.

### CODE REVIEW

**Comment by rxu** [comment]
!set Event

File: phpBB/search.php [hunk 1]
```diff
@@ -90,8 +90,21 @@
 	break;
 }
 
+$search_auth_check_override = false;
+/**
+* This event allows you to override serach auth checks
```
Comment (rubencm) [inline phpBB/search.php:95]:
```suggestion
* This event allows you to override search auth checks
```

**Comment by rubencm** [comment]
If I don't understand wrong, if you override the variable, you can bypass this check even if search is disabled.
I think what the user wants to do, can be done by using `core.common` and doing the logic there.

**Comment by rxu** [comment]
> If I don't understand wrong, if you override the variable, you can bypass this check even if search is disabled. I think what the user wants to do, can be done by using `core.common` and doing the logic there.

Not just bypass but f.e. perform additional checks which can't be done there.

**Comment by rubencm** [comment]
I think it should be possible, if the file is `search.php` just do anything you want like redirect the user to login if is anonymous

**Comment by rxu** [comment]
There's an example in the topic linked in the ticket.

**Comment by rubencm** [comment]
I have talked with @rxu in private, and this can be done using user setup events too, but this way is more specific.
Both ways are valid.

**Review by marc1706 — APPROVED** [review]

**Pull request merged**